pkg/cmd/adm/register_member.go

```diff
@@ -29,93 +39,131 @@ const (
 	AddClusterScriptURL    = AddClusterScriptDomain + AddClusterScriptPath
 )
 
+// newClientFromRestConfigFunc is a function to create a new Kubernetes client using the provided
+// rest configuration.
+type newClientFromRestConfigFunc func(*rest.Config) (runtimeclient.Client, error)
+
+// This is an extended version of the CommandContext that is used specifically just in the register member command.
+type extendedCommandContext struct {
+	*clicontext.CommandContext
+	NewClientFromRestConfig newClientFromRestConfigFunc
+}
+
+func newExtendedCommandContext(term ioutils.Terminal, clientCtor newClientFromRestConfigFunc) *extendedCommandContext {
+	return &extendedCommandContext{
+		CommandContext:          clicontext.NewCommandContext(term, nil),
+		NewClientFromRestConfig: clientCtor,
+	}
+}
+
+type registerMemberArgs struct {
+	hostKubeconfig   string
+	memberKubeconfig string
+	hostNamespace    string
+	memberNamespace  string
+	useLetsEncrypt   bool
+	memberOrdinal    *uint
+}
+
+func newRegisterMemberArgs() registerMemberArgs {
+	args := registerMemberArgs{}
+
+	defaultKubeconfigPath := ""
+	if home := homedir.HomeDir(); home != "" {
+		defaultKubeconfigPath = filepath.Join(home, ".kube", "config")
+	}
+
+	args.hostKubeconfig = defaultKubeconfigPath
+	args.memberKubeconfig = defaultKubeconfigPath
+	args.hostNamespace = "toolchain-host-operator"
+	args.memberNamespace = "toolchain-member-operator"
+	args.useLetsEncrypt = true
+
+	return args
+}
+
+type optionalUint struct {
+	*uint
+}
+
+// Set(), String() and Type() are needed to make the optionalUint a custom flag type
```

Comment:
Yes, the reason was that it is a representation of `*uint`, I.e. an optional uint argument where I needed to distinguish between "not specified" and any other legal value.

But I guess once I modify the logic for the invocations to be idempotent, as you point out elsewhere, this will not be needed.

Reply:
I took this out and replaced the parameter with just `--name-suffix` string param.